vic/drivers/shared_image/src/vic_restore.c

```diff
@@ -0,0 +1,1009 @@
+        for (k = 0; k < options.SNOW_BAND; k++) {
+            d5start[1] = k;
+            for (j = 0; j < options.Nnode; j++) {
+                d5start[2] = j;
+                get_scatter_nc_field_double(filenames.init_state,
+                                            "Soil_node_temp",
+                                            d5start, d5count, dvar);
+                for (i = 0; i < local_domain.ncells_active; i++) {
+                    v = veg_con_map[i].vidx[m];
+                    if (v >= 0) {
+                        all_vars[i].energy[v][k].T[j] = dvar[i];
+                    }
+                }
+            }
+        }
+    }
+
+    if (options.LAKES) {
+        // total soil moisture
+        for (j = 0; j < options.Nlayer; j++) {
+            d3start[0] = j;
+            get_scatter_nc_field_double(filenames.init_state,
+                                        "Lake_soil_moisture",
+                                        d3start, d3count, dvar);
+            for (i = 0; i < local_domain.ncells_active; i++) {
+                all_vars[i].lake_var.soil.layer[j].moist = dvar[i];
+            }
+        }
+
+        // ice content
+        for (j = 0; j < options.Nlayer; j++) {
+            d4start[0] = j;
+            for (p = 0; p < options.Nfrost; p++) {
+                d4start[1] = p;
+                get_scatter_nc_field_double(filenames.init_state,
+                                            "Lake_soil_ice",
+                                            d4start, d4count, dvar);
+                for (i = 0; i < local_domain.ncells_active; i++) {
+                    all_vars[i].lake_var.soil.layer[j].ice[p] = dvar[i];
+                }
+            }
+        }
+
+        if (options.CARBON) {
+            // litter carbon: tmpval = lake_var.soil.CLitter;
+            get_scatter_nc_field_double(filenames.init_state,
+                                        "Lake_CLitter",
+                                        d2start, d2count, dvar);
+            for (i = 0; i < local_domain.ncells_active; i++) {
+                all_vars[i].lake_var.soil.CLitter = dvar[i];
+            }
+
+            // intermediate carbon: tmpval = lake_var.soil.CInter;
+            get_scatter_nc_field_double(filenames.init_state,
+                                        "Lake_CInter",
+                                        d2start, d2count, dvar);
+            for (i = 0; i < local_domain.ncells_active; i++) {
+                all_vars[i].lake_var.soil.CInter = dvar[i];
+            }
+
+            // slow carbon: tmpval = lake_var.soil.CSlow;
+            get_scatter_nc_field_double(filenames.init_state,
+                                        "Lake_CSlow",
+                                        d2start, d2count, dvar);
+            for (i = 0; i < local_domain.ncells_active; i++) {
+                all_vars[i].lake_var.soil.CSlow = dvar[i];
+            }
+        }
+
+        // snow age: lake_var.snow.last_snow
```

Comment:
Oops, it already *is* in an if block, just a long one whose start is several lines above.